utils/test_results.py

```diff
@@ -59,15 +58,201 @@ class GENERATE_TEST_RESULT_PARAM:
     SKIPPED = "skipped"
 
 
+@dataclass
+class TestResultsQuery:
+    query: str
+    params: dict[str, int | str | tuple[str, ...]]
+
+
+def convert_tuple_or_none(value: set[str] | list[str] | None) -> tuple[str, ...] | None:
+    return tuple(value) if value else None
+
+
+def encode_after_or_before(value: str | None) -> str | None:
+    return str(b64decode(value.encode("ascii"))) if value else None
+
+
+def generate_base_query(
+    repoid: int,
+    ordering: tuple[str, ...],
+    ordering_direction: OrderingDirection,
+    first: int | None,
+    after: str | None,
+    last: int | None,
+    before: str | None,
+    branch: str | None,
+    interval_num_days: int,
+    testsuites: list[str] | None = None,
+    term: str | None = None,
+    test_ids: set[str] | None = None,
+) -> TestResultsQuery:
+    page_size = first or last
+
+    term_filter = f"%{term}%" if term else None
+
+    if interval_num_days not in {1, 7, 30}:
```

Comment:
nit: Im kinda going back and forth with moving all this validation logic into a separate "validate_params" fn or similar to kinda clean up this function a bit